### ISSUE TITLE
Running CMake Tools always results in an error `[driver] Invalid value to convert to cmake value:`

### ISSUE THREAD

**Issue by Danaozhong** [microsoft/vscode-cmake-tools#3931]
### Brief Issue Summary

Hi everyone,
Since yesterday I face a strange issue in my VSC CMake Tools extension.
Before:
- CMake (generate + build) worked when running from PS
- CMake Tools (VSC plugin) worked as well.

Yesterday my VSC crashed and since then, building with CMake tools from within VSC always fails with the same error message:

```
[main] Configuring project: cpp-cmake-bare 
[driver] Invalid value to convert to cmake value: {"name":"Visual Studio 16 2019","platform":"x64","toolset":"v142"}
[driver] Failed to configure project
```
This error appears in all CMake projects, not only this one.

Other commands also fail, such as `CMake: Clean`.
```
[main] Building folder: c:/Work/CPP/cpp-cmake-bare/build clean
[main] Configuring project: cpp-cmake-bare 
[driver] Invalid value to convert to cmake value: {"name":"Visual Studio 16 2019","platform":"x64","toolset":"v142"}
[driver] Failed to configure project
```
I don't even have Visual Studio 2019 installed, so this is very strange. Even if I try changing the generator tool to GCC-mingw32 it shows the same error.

Running CMake from the powershell still works.

I tried reinstalling the CMake Tools VSC extension, but the problem remains the same.

I tried setting "cmake.loggingLevel" to debug, but I got an error message:
`Unable to write into user settings. Please open the user settings to correct errors/warnings in it and try again.`

I got the feeling that somehow the CMake Tool configuration settings got corrupted/invalid. Do you have any suggestion where to look at?

### CMake Tools Diagnostics

```shell
{
  "os": "win32",
  "vscodeVersion": "1.81.0",
  "cmtVersion": "1.18.43",
  "configurations": [
    {
      "folder": "c:\\Work\\CPP\\cpp-cmake-bare",
      "cmakeVersion": "3.27.0",
      "configured": false,
      "generator": "Visual Studio 17 2022",
      "usesPresets": false,
      "compilers": {}
    }
  ],
  "cpptoolsIntegration": {
    "isReady": false,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": true
    }
  ]
}
```

**Comment by v-frankwang** [comment]
@Danaozhong I'm very sorry that we created a simple project to verify this issue, but I can't reproduce it, could you please provide the project where you reproduced this issue for the first time? This will make it easier for us to investigate this issue and I am very much looking forward to your reply!

ENV：
CMake tools: v1.18.43
vscode: v1.91.1
C/C++: v1.21.4

Actual result:
![image](https://github.com/user-attachments/assets/a59a78ac-1fd9-4a60-91f2-2ab50528562e)

**Comment by Danaozhong** [comment]
Hi @v-frankwang, thank you so much for your feedback! In my case, the problem appeared with any CMake project, even a simple hello world application. 

I did some debugging and found that for some reason, my Visual Studio Code configuration file contained invalid values:
I had to open `<user_dir>\AppData\Roaming\Code\User\settings.json`, and delete all configuration related to `cmake`.

Afterwards, CMake worked again from within Visual Studio Code and my issue is resolved! 🙇 Feel free to close this issue.

It surprised me, though, that when googling for this error message I couldn't find any hits, so hopefully this thread may help others that face a similar issue.
